site/pages/api/duplicate-consequence.api.ts

```diff
@@ -0,0 +1,65 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { getDisruptionById, upsertConsequence } from "../../data/dynamo";
+import { duplicateConsequenceSchema } from "../../schemas/consequence.schema";
+import { redirectTo, redirectToError } from "../../utils/apiUtils";
+import { getSession } from "../../utils/apiUtils/auth";
+
+const duplicateConsequence = async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        const { consequenceId } = req.query;
+
+        if (!req.query.return) {
+            throw new Error("return path required");
+        }
+
+        const validatedBody = duplicateConsequenceSchema.safeParse(req.body);
+
+        if (!consequenceId) {
+            throw new Error("consequenceId is required to duplicate a consequence");
+        }
+
+        const session = getSession(req);
+
+        if (!session) {
+            throw new Error("No session found");
+        }
+
+        if (!validatedBody.success) {
+            throw new Error("disruptionId is required to duplicate a consequence");
+        }
+
+        const disruption = await getDisruptionById(validatedBody.data.disruptionId, session.orgId);
+
+        if (!disruption || !disruption.consequences) {
+            throw new Error("No disruption / disruption with consequences found");
+        }
+
+        const consequenceToDuplicate = disruption.consequences.find(
+            (consequence) => consequence.consequenceIndex === Number(consequenceId),
+        );
+
+        if (!consequenceToDuplicate) {
+            throw new Error("No consequence found to duplicate");
+        }
+
+        await upsertConsequence(
+            { ...consequenceToDuplicate, consequenceIndex: disruption.consequences.length },
```

Comment:
The consequenceIndex should be 1 plus the largest index in consequences array. Use of length does not consider deleted consequences. 

For example Initially they could be 3 consequences added and then the first consequence is deleted. Using length would mean that it would update the last consequence instead of creating a new one.

Reply:
done